Kuroba/app/src/main/java/com/github/adamantcheese/chan/core/cache/stream/WebmStreamingSource.kt

```diff
@@ -110,6 +96,164 @@ class WebmStreamingSource(
         cancelableDownload.stop()
     }
 
+    private fun startLoadingFromNetwork(
+            file: AbstractFile,
+            fileCacheSource: WebmStreamingDataSource,
+            callback: MediaSourceCallback,
+            uri: Uri
+    ) {
+        // The webm file is either partially downloaded or is not downloaded at all.
+        // We take whatever there is and load it into the WebmStreamingDataSource so
+        // we don't need to redownload the bytes that have already been downloaded
+        val exists = fileManager.exists(file)
+        val fileLength = fileManager.getLength(file)
+
+        Logger.d(TAG,
+                "createMediaSource() Loading partially downloaded file after stop(), " +
+                        "fileLength = $fileLength")
+
+        if (exists && fileLength > 0L) {
+            try {
+                fileManager.getInputStream(file)?.use { inputStream ->
+                    fileCacheSource.fillCache(fileLength, inputStream)
+                } ?: throw IOException("Couldn't get input stream for file " +
+                        "(${file.getFullPath()})")
+            } catch (error: IOException) {
+                Logger.e(TAG, "createMediaSource() Failed to fill cache!", error)
+            }
+        }
+
+        callback.onMediaSourceReady(
+                ProgressiveMediaSource.Factory(DataSource.Factory { fileCacheSource })
+                        .createMediaSource(uri)
+        )
+    }
+
+    private fun loadFromCacheFile(rawFile: RawFile, callback: MediaSourceCallback) {
+        Logger.d(TAG, "createMediaSource() Loading already downloaded file from the disk")
+        val fileUri = Uri.parse(rawFile.getFullPath())
+
+        callback.onMediaSourceReady(
+                ProgressiveMediaSource.Factory(DataSource.Factory { FileDataSource() })
+                        .createMediaSource(fileUri)
+        )
+    }
+
+    private fun loadFromLocalThread(
```

Comment:
This seems remarkably similar to FileCacheV2's handleLocalThreadFile, maybe the two can be combined into one?

Reply:
Same.